src/main/java/uk/gov/justice/hmpps/prison/service/BedAssignmentHistoryService.java

```diff
@@ -106,7 +106,7 @@ public List<BedAssignment> getBedAssignmentsHistoryByDateForAgency(final String
     }
 
     private BedAssignment transform(final BedAssignmentHistory assignment) {
-        final var agencyInternalLocation = locationRepository.findOneByLocationId(assignment.getLivingUnitId());
```

Comment:
already loaded in the assignment records so no need for the additional repo lookup